internal/systemd.go

```diff
@@ -0,0 +1,30 @@
+package internal
+
+import (
+	"os/exec"
+	"strings"
+)
+
+// IsServiceActive check if given service is active
+func IsServiceActive(service string) bool {
+	out, err := exec.Command(SystemctlExec, "is-active", service).Output()
+	if err != nil {
+		return false
+	}
+	return "active" == strings.Trim(strings.Trim(string(out), "\n"), " ")
+}
+
+// IsSystemShutdown detect if system is being shutdown
+func IsSystemShutdown() bool {
+	return FileExists("/run/nologin") || FileExists("/var/run/nologin")
+}
+
+// IsSystemd detect if system is running systemd
+func IsSystemd() bool {
+	// check name of PID1 process: "ps -p 1 -o comm="
+	out, err := exec.Command("ps", "-p", "1", "-o", "comm=").Output()
```

Comment:
According to systemd documentation. the proper way to check whether the system was booted up using the systemd is to check the presence of ` /run/systemd/system` directory: https://www.freedesktop.org/software/systemd/man/latest/sd_booted.html

Reply:
Thx, good to know, indeed, this way is simpler.

Reply:
Done